File: datamodelgov-examples/python-client-example/simple_example.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
DataModelGov Python SDK 客户端简单示例
只演示SDK的使用，不依赖其他模块
"""

import json
import logging
import time
from datetime import datetime
from typing import Dict, List, Any, Optional

# 配置日志
logging.basicConfig(
    level=logging.INFO,
    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s'
)
logger = logging.getLogger(__name__)


class DataModelGovClient:
    """DataModelGov Python SDK 客户端示例"""
    
    def __init__(self, base_url: str = "http://localhost:8080"):
        self.base_url = base_url
        # 这里应该初始化 DataModelGov SDK
        # 示例代码：
        # from datamodelgov import DataModelGovSDK
        # self.sdk = DataModelGovSDK(base_url)
        logger.info(f"DataModelGov 客户端初始化完成: {base_url}")
    
    def create_model(self, model_data: Dict[str, Any]) -> Dict[str, Any]:
        """创建数据模型示例"""
        logger.info(f"SDK调用: create_model() - 创建数据模型: {model_data}")
        
        # 这里应该调用 DataModelGov SDK 创建数据模型
        # 示例代码：
        # result = self.sdk.create_model(model_data)
        
        # 模拟创建成功
        response = {
            "id": f"model_{int(time.time())}",
            "name": model_data.get("name"),
            "description": model_data.get("description"),
            "fields": model_data.get("fields", {}),
            "createdAt": datetime.now().isoformat(),
            "updatedAt": datetime.now().isoformat()
        }
        
        return response
    
    def get_models(self, name: Optional[str] = None) -> List[Dict[str, Any]]:
        """查询数据模型示例"""
        logger.info(f"SDK调用: get_models() - 查询数据模型: name={name}")
        
        # 这里应该调用 DataModelGov SDK 查询数据模型
        # 示例代码：
        # result = self.sdk.get_models(name)
        
        # 模拟查询结果
        models = [
            {
                "id": "model1",
                "name": "用户模型",
                "description": "用户数据模型",
                "fields": {"name": "string", "age": "int"},
                "createdAt": datetime.now().isoformat(),
                "updatedAt": datetime.now().isoformat()
            },
            {
                "id": "model2",
                "name": "订单模型",
                "description": "订单数据模型",
                "fields": {"orderId": "string", "amount": "float"},
                "createdAt": datetime.now().isoformat(),
                "updatedAt": datetime.now().isoformat()
            }
        ]
        
        return models
    
    def get_model(self, model_id: str) -> Dict[str, Any]:
        """获取单个模型示例"""
        logger.info(f"SDK调用: get_model() - 获取单个模型: model_id={model_id}")
        
        # 这里应该调用 DataModelGov SDK 获取单个模型
        # 示例代码：
        # result = self.sdk.get_model(model_id)
        
        # 模拟获取结果
        model = {
            "id": model_id,
            "name": "示例模型",
            "description": "这是一个示例模型",
            "fields": {"id": "string", "name": "string"},
            "createdAt": datetime.now().isoformat(),
            "updatedAt": datetime.now().isoformat()
        }
        
        return model
    
    def update_model(self, model_id: str, model_data: Dict[str, Any]) -> Dict[str, Any]:
        """更新数据模型示例"""
        logger.info(f"SDK调用: update_model() - 更新数据模型: model_id={model_id}, model_data={model_data}")
        
        # 这里应该调用 DataModelGov SDK 更新数据模型
        # 示例代码：
        # result = self.sdk.update_model(model_id, model_data)
        
        # 模拟更新成功
        response = {
            "id": model_id,
            "name": model_data.get("name"),
            "description": model_data.get("description"),
            "fields": model_data.get("fields", {}),
            "createdAt": datetime.now().isoformat(),
            "updatedAt": datetime.now().isoformat()
        }
        
        return response
    
    def delete_model(self, model_id: str) -> bool:
        """删除数据模型示例"""
        logger.info(f"SDK调用: delete_model() - 删除数据模型: model_id={model_id}")
        
        # 这里应该调用 DataModelGov SDK 删除数据模型
        # 示例代码：
        # result = self.sdk.delete_model(model_id)
        
        # 模拟删除成功
        return True
    
    def analyze_data(self, analysis_request: Dict[str, Any]) -> Dict[str, Any]:
        """数据分析示例"""
        logger.info(f"SDK调用: analyze_data() - 数据分析: {analysis_request}")
        
        # 这里应该调用 DataModelGov SDK 进行数据分析
        # 示例代码：
        # result = self.sdk.analyze_data(analysis_request)
        
        model_id = analysis_request.get("modelId")
        data = analysis_request.get("data", [])
        
        response = {
            "analysisId": f"analysis_{int(time.time())}",
            "modelId": model_id,
            "summary": "数据分析完成",
            "totalRecords": len(data) // 2,  # 假设每两条数据代表一个记录
            "analysisTime": datetime.now().isoformat(),
            "details": {
                "processedRecords": len(data) // 2,
                "analysisType": "statistical"
            }
        }
        
        return response


def demonstrate_json_serialization():
    """演示JSON序列化"""
    print("=== JSON序列化示例 ===")
    
    model = {
        "name": "测试模型",
        "description": "JSON序列化测试",
        "fields": {"test": "string"},
        "createdAt": datetime.now().isoformat(),
        "updatedAt": datetime.now().isoformat()
    }
    
    # 序列化为JSON
    json_data = json.dumps(model, indent=2, ensure_ascii=False)
    print(f"模型JSON:\n{json_data}\n")
    
    # 从JSON反序列化
    parsed_model = json.loads(json_data)
    print(f"反序列化成功: {parsed_model['name']}")


def main():
    """主函数 - 演示所有SDK调用"""
    print("=== DataModelGov Python SDK 客户端示例 ===\n")
    
    # 1. 创建客户端
    print("1. 创建客户端")
    client = DataModelGovClient("http://localhost:8080")
    print("SDK调用: client = DataModelGovClient(\"http://localhost:8080\")")
    print("客户端创建成功\n")
    
    # 2. 创建数据模型
    print("2. 创建数据模型")
    model_data = {
        "name": "用户模型",
        "description": "用户数据模型示例",
        "fields": {
            "name": "string",
            "age": "int",
            "email": "string"
        }
    }
    
    print("SDK调用: client.create_model(model_data)")
    created_model = client.create_model(model_data)
    print(f"创建成功: {created_model['id']}")
    print(f"模型名称: {created_model['name']}")
    print(f"模型描述: {created_model['description']}")
    print()
    
    # 3. 查询数据模型
    print("3. 查询数据模型")
    print("SDK调用: client.get_models()")
    models = client.get_models()
    print(f"查询到 {len(models)} 个模型:")
    for m in models:
        print(f"  - {m['name']} ({m['id']})")
    print()
    
    # 4. 按名称查询模型
    print("4. 按名称查询模型")
    print("SDK调用: client.get_models(\"用户模型\")")
    user_models = client.get_models("用户模型")
    print(f"查询到 {len(user_models)} 个用户模型")
    print()
    
    # 5. 获取单个模型
    print("5. 获取单个模型")
    print(f"SDK调用: client.get_model(\"{created_model['id']}\")")
    single_model = client.get_model(created_model['id'])
    print("模型详情:")
    print(f"  ID: {single_model['id']}")
    print(f"  名称: {single_model['name']}")
    print(f"  描述: {single_model['description']}")
    print(f"  字段: {single_model['fields']}")
    print()
    
    # 6. 更新数据模型
    print("6. 更新数据模型")
    update_data = {
        "name": "更新用户模型",
        "description": "更新后的用户数据模型",
        "fields": {
            "name": "string",
            "age": "int",
            "email": "string",
            "phone": "string"
        }
    }
    
    print(f"SDK调用: client.update_model(\"{created_model['id']}\", update_data)")
    updated_model = client.update_model(created_model['id'], update_data)
    print(f"更新成功: {updated_model['name']}")
    print()
    
    # 7. 数据分析
    print("7. 数据分析")
    analysis_request = {
        "modelId": created_model['id'],
        "data": [
            "张三", "25", "zhangsan@example.com", "13800138000",
            "李四", "30", "lisi@example.com", "13800138001"
        ]
    }
    
    print(f"SDK调用: client.analyze_data(analysis_request)")
    result = client.analyze_data(analysis_request)
    print("分析结果:")
    print(f"  分析ID: {result['analysisId']}")
    print(f"  模型ID: {result['modelId']}")
    print(f"  摘要: {result['summary']}")
    print(f"  总记录数: {result['totalRecords']}")
    print(f"  分析时间: {result['analysisTime']}")
    print()
    
    # 8. 删除数据模型
    print("8. 删除数据模型")
    print(f"SDK调用: client.delete_model(\"{created_model['id']}\")")
    success = client.delete_model(created_model['id'])
    print(f"删除成功: {success}")
    print()
    
    print("=== 所有SDK调用演示完成 ===")
    print()
    
    # 9. JSON序列化示例
    demonstrate_json_serialization()


if __name__ == "__main__":
    main()

File: datamodelgov-examples/go-client-example/simple_example.go
package main

import (
	"encoding/json"
	"fmt"
	"log"
	"time"
)

// Model 数据模型结构
type Model struct {
	ID          string            `json:"id"`
	Name        string            `json:"name"`
	Description string            `json:"description"`
	Fields      map[string]string `json:"fields"`
	CreatedAt   time.Time         `json:"createdAt"`
	UpdatedAt   time.Time         `json:"updatedAt"`
}

// AnalysisResult 分析结果结构
type AnalysisResult struct {
	AnalysisID   string                 `json:"analysisId"`
	ModelID      string                 `json:"modelId"`
	Summary      string                 `json:"summary"`
	TotalRecords int                    `json:"totalRecords"`
	AnalysisTime time.Time              `json:"analysisTime"`
	Details      map[string]interface{} `json:"details"`
}

// DataModelGovClient DataModelGov Go SDK 客户端示例
type DataModelGovClient struct {
	BaseURL string
}

// NewDataModelGovClient 创建新的客户端
func NewDataModelGovClient(baseURL string) *DataModelGovClient {
	return &DataModelGovClient{
		BaseURL: baseURL,
	}
}

// CreateModel 创建数据模型示例
func (c *DataModelGovClient) CreateModel(model *Model) (*Model, error) {
	// 这里应该调用 DataModelGov SDK 创建数据模型
	// 示例代码：
	// sdkClient := datamodelgov.NewClient(c.BaseURL)
	// result, err := sdkClient.CreateModel(model)
	
	log.Printf("SDK调用: CreateModel() - 创建数据模型: %+v", model)
	
	// 模拟创建成功
	createdModel := &Model{
		ID:          fmt.Sprintf("model_%d", time.Now().Unix()),
		Name:        model.Name,
		Description: model.Description,
		Fields:      model.Fields,
		CreatedAt:   time.Now(),
		UpdatedAt:   time.Now(),
	}
	
	return createdModel, nil
}

// GetModels 查询数据模型示例
func (c *DataModelGovClient) GetModels(name string) ([]*Model, error) {
	// 这里应该调用 DataModelGov SDK 查询数据模型
	// 示例代码：
	// sdkClient := datamodelgov.NewClient(c.BaseURL)
	// result, err := sdkClient.GetModels(name)
	
	log.Printf("SDK调用: GetModels() - 查询数据模型: name=%s", name)
	
	// 模拟查询结果
	models := []*Model{
		{
			ID:          "model1",
			Name:        "用户模型",
			Description: "用户数据模型",
			Fields:      map[string]string{"name": "string", "age": "int"},
			CreatedAt:   time.Now(),
			UpdatedAt:   time.Now(),
		},
		{
			ID:          "model2",
			Name:        "订单模型",
			Description: "订单数据模型",
			Fields:      map[string]string{"orderId": "string", "amount": "float"},
			CreatedAt:   time.Now(),
			UpdatedAt:   time.Now(),
		},
	}
	
	return models, nil
}

// GetModel 获取单个模型示例
func (c *DataModelGovClient) GetModel(modelID string) (*Model, error) {
	// 这里应该调用 DataModelGov SDK 获取单个模型
	// 示例代码：
	// sdkClient := datamodelgov.NewClient(c.BaseURL)
	// result, err := sdkClient.GetModel(modelID)
	
	log.Printf("SDK调用: GetModel() - 获取单个模型: modelID=%s", modelID)
	
	// 模拟获取结果
	model := &Model{
		ID:          modelID,
		Name:        "示例模型",
		Description: "这是一个示例模型",
		Fields:      map[string]string{"id": "string", "name": "string"},
		CreatedAt:   time.Now(),
		UpdatedAt:   time.Now(),
	}
	
	return model, nil
}

// UpdateModel 更新数据模型示例
func (c *DataModelGovClient) UpdateModel(modelID string, model *Model) (*Model, error) {
	// 这里应该调用 DataModelGov SDK 更新数据模型
	// 示例代码：
	// sdkClient := datamodelgov.NewClient(c.BaseURL)
	// result, err := sdkClient.UpdateModel(modelID, model)
	
	log.Printf("SDK调用: UpdateModel() - 更新数据模型: modelID=%s, model=%+v", modelID, model)
	
	// 模拟更新成功
	updatedModel := &Model{
		ID:          modelID,
		Name:        model.Name,
		Description: model.Description,
		Fields:      model.Fields,
		CreatedAt:   time.Now(),
		UpdatedAt:   time.Now(),
	}
	
	return updatedModel, nil
}

// DeleteModel 删除数据模型示例
func (c *DataModelGovClient) DeleteModel(modelID string) error {
	// 这里应该调用 DataModelGov SDK 删除数据模型
	// 示例代码：
	// sdkClient := datamodelgov.NewClient(c.BaseURL)
	// err := sdkClient.DeleteModel(modelID)
	
	log.Printf("SDK调用: DeleteModel() - 删除数据模型: modelID=%s", modelID)
	
	// 模拟删除成功
	return nil
}

// AnalyzeData 数据分析示例
func (c *DataModelGovClient) AnalyzeData(modelID string, data []string) (*AnalysisResult, error) {
	// 这里应该调用 DataModelGov SDK 进行数据分析
	// 示例代码：
	// sdkClient := datamodelgov.NewClient(c.BaseURL)
	// result, err := sdkClient.AnalyzeData(modelID, data)
	
	log.Printf("SDK调用: AnalyzeData() - 数据分析: modelID=%s, data=%v", modelID, data)
	
	// 模拟分析结果
	result := &AnalysisResult{
		AnalysisID:   fmt.Sprintf("analysis_%d", time.Now().Unix()),
		ModelID:      modelID,
		Summary:      "数据分析完成",
		TotalRecords: len(data) / 2, // 假设每两条数据代表一个记录
		AnalysisTime: time.Now(),
		Details: map[string]interface{}{
			"processedRecords": len(data) / 2,
			"analysisType":     "statistical",
		},
	}
	
	return result, nil
}

// 演示JSON序列化
func demonstrateJSONSerialization() {
	fmt.Println("=== JSON序列化示例 ===")
	
	model := &Model{
		Name:        "测试模型",
		Description: "JSON序列化测试",
		Fields:      map[string]string{"test": "string"},
		CreatedAt:   time.Now(),
		UpdatedAt:   time.Now(),
	}
	
	// 序列化为JSON
	jsonData, err := json.Marshal(model)
	if err != nil {
		log.Printf("JSON序列化失败: %v", err)
		return
	}
	
	fmt.Printf("模型JSON:\n%s\n\n", string(jsonData))
	
	// 从JSON反序列化
	var parsedModel Model
	err = json.Unmarshal(jsonData, &parsedModel)
	if err != nil {
		log.Printf("JSON反序列化失败: %v", err)
		return
	}
	
	fmt.Printf("反序列化成功: %s\n", parsedModel.Name)
}

func main() {
	fmt.Println("=== DataModelGov Go SDK 客户端示例 ===\n")
	
	// 1. 创建客户端
	fmt.Println("1. 创建客户端")
	client := NewDataModelGovClient("http://localhost:8080")
	fmt.Println("SDK调用: client := NewDataModelGovClient(\"http://localhost:8080\")")
	fmt.Println("客户端创建成功\n")
	
	// 2. 创建数据模型
	fmt.Println("2. 创建数据模型")
	model := &Model{
		Name:        "用户模型",
		Description: "用户数据模型示例",
		Fields: map[string]string{
			"name":  "string",
			"age":   "int",
			"email": "string",
		},
	}
	
	fmt.Println("SDK调用: client.CreateModel(model)")
	createdModel, err := client.CreateModel(model)
	if err != nil {
		log.Printf("创建模型失败: %v", err)
		return
	}
	fmt.Printf("创建成功: %s\n", createdModel.ID)
	fmt.Printf("模型名称: %s\n", createdModel.Name)
	fmt.Printf("模型描述: %s\n", createdModel.Description)
	fmt.Println()
	
	// 3. 查询数据模型
	fmt.Println("3. 查询数据模型")
	fmt.Println("SDK调用: client.GetModels()")
	models, err := client.GetModels("")
	if err != nil {
		log.Printf("查询模型失败: %v", err)
		return
	}
	fmt.Printf("查询到 %d 个模型:\n", len(models))
	for _, m := range models {
		fmt.Printf("  - %s (%s)\n", m.Name, m.ID)
	}
	fmt.Println()
	
	// 4. 按名称查询模型
	fmt.Println("4. 按名称查询模型")
	fmt.Println("SDK调用: client.GetModels(\"用户模型\")")
	userModels, err := client.GetModels("用户模型")
	if err != nil {
		log.Printf("按名称查询模型失败: %v", err)
		return
	}
	fmt.Printf("查询到 %d 个用户模型\n", len(userModels))
	fmt.Println()
	
	// 5. 获取单个模型
	fmt.Println("5. 获取单个模型")
	fmt.Printf("SDK调用: client.GetModel(\"%s\")\n", createdModel.ID)
	singleModel, err := client.GetModel(createdModel.ID)
	if err != nil {
		log.Printf("获取单个模型失败: %v", err)
		return
	}
	fmt.Println("模型详情:")
	fmt.Printf("  ID: %s\n", singleModel.ID)
	fmt.Printf("  名称: %s\n", singleModel.Name)
	fmt.Printf("  描述: %s\n", singleModel.Description)
	fmt.Printf("  字段: %v\n", singleModel.Fields)
	fmt.Println()
	
	// 6. 更新数据模型
	fmt.Println("6. 更新数据模型")
	updateModel := &Model{
		Name:        "更新用户模型",
		Description: "更新后的用户数据模型",
		Fields: map[string]string{
			"name":  "string",
			"age":   "int",
			"email": "string",
			"phone": "string",
		},
	}
	
	fmt.Printf("SDK调用: client.UpdateModel(\"%s\", updateModel)\n", createdModel.ID)
	updatedModel, err := client.UpdateModel(createdModel.ID, updateModel)
	if err != nil {
		log.Printf("更新模型失败: %v", err)
		return
	}
	fmt.Printf("更新成功: %s\n", updatedModel.Name)
	fmt.Println()
	
	// 7. 数据分析
	fmt.Println("7. 数据分析")
	data := []string{
		"张三", "25", "zhangsan@example.com", "13800138000",
		"李四", "30", "lisi@example.com", "13800138001",
	}
	
	fmt.Printf("SDK调用: client.AnalyzeData(\"%s\", data)\n", createdModel.ID)
	result, err := client.AnalyzeData(createdModel.ID, data)
	if err != nil {
		log.Printf("数据分析失败: %v", err)
		return
	}
	fmt.Println("分析结果:")
	fmt.Printf("  分析ID: %s\n", result.AnalysisID)
	fmt.Printf("  模型ID: %s\n", result.ModelID)
	fmt.Printf("  摘要: %s\n", result.Summary)
	fmt.Printf("  总记录数: %d\n", result.TotalRecords)
	fmt.Printf("  分析时间: %s\n", result.AnalysisTime.Format("2006-01-02 15:04:05"))
	fmt.Println()
	
	// 8. 删除数据模型
	fmt.Println("8. 删除数据模型")
	fmt.Printf("SDK调用: client.DeleteModel(\"%s\")\n", createdModel.ID)
	err = client.DeleteModel(createdModel.ID)
	if err != nil {
		log.Printf("删除模型失败: %v", err)
		return
	}
	fmt.Println("删除成功")
	fmt.Println()
	
	fmt.Println("=== 所有SDK调用演示完成 ===")
	fmt.Println()
	
	// 9. JSON序列化示例
	demonstrateJSONSerialization()
}
